tools/extract-new-translation-keys/extract-new-translation-keys.mjs

```diff
@@ -0,0 +1,226 @@
+      );
+
+      const messageEntry = await createPluralMessageEntry(client, languageValues, count);
+      await createTranslationEntry(client, key, description, messageEntry);
+    } else {
+      const messageEntry = await createMessageEntry(client, { 'en-US': poItem.msgstr[0] });
+      await createTranslationEntry(client, key, description, messageEntry);
+    }
+  }
+}
+
+async function runCommand(command, message) {
+  try {
+    process.stdout.write(`${message}...`);
+    const result = await command();
+    console.log(' DONE');
+    return result;
+  } catch (error) {
+    console.error(`\nError: ${error}`);
+    process.exit(1);
+  }
+}
+
+async function fetchUpstreamTranslationKeys() {
+  return await fetch(CATALOG_URL).then(async (response) => {
+    if (!response.ok) {
+      return Promise.reject(
+        `Fetching catalog from ${CATALOG_URL} failed with ${response.status} ${response.statusText}`
+      );
+    }
+
+    return new Set(Object.keys(await response.json()));
+  });
+}
+
+async function extractNewTranslations(upstreamTranslationKeys) {
+  const extractTranslationKeys = spawnSync('npm', ['run', 'extract-translation-keys']);
+  if (extractTranslationKeys.status !== 0) {
+    throw new Error(`Extracting keys failed with ${extractTranslationKeys.stderr.toString()}`);
+  }
+
+  const po = await readCatalog(catalogPath);
+  return po.items.filter((item) => !upstreamTranslationKeys.has(item.msgid));
+}
+
+function readCatalog(path) {
+  return new Promise((resolve, reject) => {
+    PO.load(path, (err, po) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(po);
+      }
+    });
+  });
+}
+
+async function createMessageEntry(client, languageValue) {
+  return await client.entry.create(
+    {
+      contentTypeId: 'message',
+      spaceId: SPACE_ID,
+      environmentId: ENVIRONMENT_ID,
+    },
+    {
+      fields: { languageValue },
+    }
+  );
+}
+
+async function createPluralMessageEntry(client, languageValues, count) {
+  return await client.entry.create(
+    {
+      contentTypeId: 'messageWithPluralization',
+      spaceId: SPACE_ID,
+      environmentId: ENVIRONMENT_ID,
+    },
+    {
+      fields: {
+        count: { 'en-US': count },
+        ...languageValues,
+      },
+    }
+  );
+}
+
+async function createTranslationEntry(client, key, description, messageEntry) {
+  return client.entry
+    .create(
+      {
+        contentTypeId: 'translation',
+        spaceId: SPACE_ID,
+        environmentId: ENVIRONMENT_ID,
+      },
+      {
+        fields: {
+          translationKey: { 'en-US': key },
+          description: { 'en-US': description },
+          message: {
+            'en-US': {
+              sys: {
+                type: 'Link',
+                linkType: 'Entry',
+                id: messageEntry.sys.id,
+              },
+            },
+          },
+        },
+      }
+    )
+    // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
```

Comment:
What’s unused here? 🤔

Reply:
👀 that was before the change, _entry was not used, i remove the comment. thanks for pointing that out